mtp_api/apps/transaction/tests/test_views.py

```diff
@@ -7,105 +7,744 @@
+        self._request_and_assert(
+            status_param=TRANSACTION_STATUS.AVAILABLE
+        )
+
+    def test_with_credited_status(self):
+        self._request_and_assert(
+            status_param=TRANSACTION_STATUS.CREDITED
+        )
+
+
+class TransactionsTakeTestCase(BaseTransactionViewTestCase):
+
+    def _get_url(self, user, prison, count=None):
+        url = reverse(
+            'transaction-prison-user-take', kwargs={
+                'user_id': user.pk,
+                'prison_id': prison.pk
+            }
+        )
+
+        if count:
+            url += '?count={count}'.format(count=count)
+
+        return url
+
+    def test_take_within_limit(self):
+        prison = self.prisons[0]
+        owner = get_users_for_prison(prison)[0]
+
+        count = 1
+
+        # delete pending transactions just to clean things up
+        self._get_pending_transactions_qs(prison, owner).delete()
+
+        # check no taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, owner).count(),
+            0
+        )
+
+        # request
+        self.client.force_authenticate(user=owner)
+
+        url = self._get_url(owner, prison, count)
+        response = self.client.post(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_303_SEE_OTHER)
+
+        # check 5 taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, owner).count(),
+            count
+        )
+
+    def test_nobody_else_can_take_transactions_for_others(self):
+        """
+        Anybody managing a prison should be able to take transactions
+        on behalf of other users within the same prison.
+        """
+        prison = self.prisons[0]
+        users = get_users_for_prison(prison)
+        self.assertTrue(len(users) >= 2)
+
+        # We need 2 users as we want to test that logged_in_user
+        # can take transactions on behalf of transactions_owner.
+        # As result, the transactions.owner should be transactions_owner
+        # and not logged_in_user
+        logged_in_user = users[0]
+        transactions_owner = users[1]
+
+        # delete pending transactions just to clean things up
+        self._get_pending_transactions_qs(prison, transactions_owner).delete()
+
+        # check no taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, transactions_owner).count(),
+            0
+        )
+
+        # request
+        self.client.force_authenticate(user=logged_in_user)
+
+        url = self._get_url(transactions_owner, prison, 1)
+        response = self.client.post(url, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # check 5 taken transactions in db
+        self.assertEqual(
+            self._get_pending_transactions_qs(prison, transactions_owner).count(),
+            0
```

Comment:
Should this be asserting against something other than 0? (The comment suggests it should be 5?)